### PR TITLE
OCPQE-24306: Add agent gather

### PR DESCRIPTION
**Pull request by mhanss**
This PR aims to add support for agent gather, which is essential for troubleshooting ABI failures.

### CODE REVIEW

**Comment by mhanss** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-baremetal-sno-agent-ipv4-static-connected-f7

**Comment by mhanss** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-baremetal-sno-agent-ipv4-static-connected-f7

**Comment by mhanss** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-baremetal-sno-agent-ipv4-static-connected-f7

File: ci-operator/step-registry/baremetal/lab/agent/gather/baremetal-lab-agent-gather-commands.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+node_zero_ip=$(<"${SHARED_DIR}"/node-zero-ip.txt)
+SSH_PORT=2222
+SSHOPTS=(-o 'ConnectTimeout=5'
+  -o 'StrictHostKeyChecking=no'
+  -o 'UserKnownHostsFile=/dev/null'
+  -o LogLevel=ERROR
+  -i "${CLUSTER_PROFILE_DIR}/ssh-key")
+
+echo "Trying to gather agent logs on the host ${node_zero_ip}"
+ssh "${SSHOPTS[@]}" -N -L "${SSH_PORT}":"${node_zero_ip}":22 root@"${AUX_HOST}" &
```
Comment (aleskandro) [inline ci-operator/step-registry/baremetal/lab/agent/gather/baremetal-lab-agent-gather-commands.sh:29]: Hey @mhanss let's avoid doing tunneling if possible. See https://github.com/openshift/release/pull/47240/files#diff-e8f157108c4e736639dbbeeb84d5ac1b1f9da2291a126f7313f1eb8796891895R300-R303 for how to access nodes through the AUX_HOST and the already-configured port-forwarding.

**Comment by mhanss** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-baremetal-sno-agent-dualstack-dynamic-f14

**Comment by mhanss** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetal-sno-agent-dualstack-dynamic-f14

**Comment by mhanss** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetal-sno-agent-dualstack-dynamic-f14

**Comment by mhanss** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetal-sno-agent-dualstack-dynamic-f14

**Comment by mhanss** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetal-sno-agent-dualstack-dynamic-f14

**Comment by mhanss** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetal-sno-agent-dualstack-dynamic-f14

**Comment by mhanss** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetal-sno-agent-dualstack-dynamic-f14

**Comment by mhanss** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetal-sno-agent-dualstack-dynamic-f14

**Comment by mhanss** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetal-sno-agent-dualstack-dynamic-f14

**Comment by mhanss** [comment]
Successfully collected the agent gather https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/55238/rehearse-55238-periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetal-sno-agent-dualstack-dynamic-f14/1833418245051256832/artifacts/baremetal-sno-agent-dualstack-dynamic-f14/baremetal-lab-agent-gather/artifacts/

File: ci-operator/step-registry/baremetal/lab/pre/load-balancer/baremetal-lab-pre-load-balancer-commands.sh [hunk 2]
```diff
@@ -43,6 +44,13 @@ for bmhost in $(yq e -o=j -I=0 '.[]' "${SHARED_DIR}/hosts.yaml"); do
       server $name $ip:443 check inter 1s
       server $name-v6 [$ipv6]:443 check inter 1s"
   fi
+  SSH="$SSH
```
[the remark this answers is not in the document]
Comment (mhanss) [reply ci-operator/step-registry/baremetal/lab/pre/load-balancer/baremetal-lab-pre-load-balancer-commands.sh:47]: Done

File: ci-operator/step-registry/baremetal/lab/agent/conf/dynamic/baremetal-lab-agent-conf-dynamic-commands.sh [hunk 3]
```diff
@@ -8,6 +8,7 @@ if [ "${ipv6_enabled:-}" == "true" ]; then
  QUERY=".[0].ipv6"
 fi
 RENDEZVOUS_IP="$(yq -r e -o=j -I=0 "${QUERY}" "${SHARED_DIR}/hosts.yaml")"
+yq -r e -o=j -I=0 ".[0].host" "${SHARED_DIR}/hosts.yaml" >"${SHARED_DIR}"/host-id.txt
```
[the remark this answers is not in the document]
Comment (mhanss) [reply ci-operator/step-registry/baremetal/lab/agent/conf/dynamic/baremetal-lab-agent-conf-dynamic-commands.sh:11]: No, it is added to two files `conf-dynamic-commands.sh` and `conf-static-commands.sh`.

**Comment by mhanss** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetal-sno-agent-ipv6-dynamic-f28

**Comment by mhanss** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetal-sno-agent-ipv6-dynamic-f28

**Comment by mhanss** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetal-sno-agent-ipv6-dynamic-f28

File: ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-commands.sh [hunk 4]
```diff
@@ -232,18 +233,9 @@ echo "Launching 'wait-for bootstrap-complete' installation step....."
 # Let's use a proxy here as the internal net is not routable from the container running the installer.
 http_proxy="${proxy}" https_proxy="${proxy}" HTTP_PROXY="${proxy}" HTTPS_PROXY="${proxy}" \
   oinst agent wait-for bootstrap-complete 2>&1 &
-if ! wait $!; then
-  # TODO: gather logs??
-  echo "ERROR: Bootstrap failed. Aborting execution."
-  exit 1
-fi
```
Comment (aleskandro) [inline ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-commands.sh:236]: We still need that wait.

```suggestion
if ! wait $!; then
  echo "ERROR: Bootstrap failed. Aborting execution."
  exit 1
fi
```

File: ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-commands.sh [hunk 5]
```diff
@@ -232,18 +233,9 @@ echo "Launching 'wait-for bootstrap-complete' installation step....."
...
 update_image_registry &
 echo -e "\nLaunching 'wait-for install-complete' installation step....."
 http_proxy="${proxy}" https_proxy="${proxy}" HTTP_PROXY="${proxy}" HTTPS_PROXY="${proxy}" \
   oinst agent wait-for install-complete &
-if ! wait "$!"; then
-  echo "ERROR: Installation failed. Aborting execution."
-  # TODO: gather logs??
-  exit 1
-fi
+
```
Comment (aleskandro) [inline ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-commands.sh:241]:
```suggestion
if ! wait "$!"; then
  echo "ERROR: Installation failed. Aborting execution."
  exit 1
fi
```

**Review by aleskandro — APPROVED** [review]
/lgtm
/approve

Thanks @mhanss

**Pull request merged**